### ISSUE TITLE
Allow region and projectId in Scaleway provider to be references

### ISSUE THREAD

**Issue by mig4ng** [external-secrets/external-secrets#2827]
**Is your feature request related to a problem? Please describe.**
It would be nice to configure the region and project id of the Scaleway provider to be secretRef or even configMap references. When I try to use region and project id with a secretRef I get:
```
 Cannot patch "secret-store" with kind SecretStore: SecretStore.external-secrets.io "secret-store" is invalid: [spec.provider.scaleway.projectId: Invalid value: "object": spec.provider.scaleway.projectId in body must be of type string: "object", spec.provider.scaleway.region: Invalid value: "object": spec.provider.scaleway.region in body must be of type string: "object"] 
 ```

**Describe the solution you'd like**
Allow project id and region to be set as secretRef too.

**Describe alternatives you've considered**
Right now the alternative I have is hard coding the project id (which is a random id) in the SecretStore yaml definition.
Alternatives would be setting it up with the project name or allowing it to come from a secret/configmap like the access and secret keys (since where I am creating the secret I have the project name and id).

**Additional context**
I went through the code and found this:
https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/scaleway/provider.go#L165-L171

It is expecting a string both on region and project id and validating it with Scaleway's Go SDK validation module.

**Comment by ojroques** [comment]
Hello, I'm one of the maintainers of the Scaleway provider. The change you request would be a breaking change: it would require to change the type of fields `region` and `projectID` from strings to objects. So we cannot simply implement this feature.

Could you provide more details about your use case and why you need to use a secretRef instead of a string for these fields? Maybe there are other ways we could help.

**Comment by mig4ng** [comment]
> Could you provide more details about your use case and why you need to use a secretRef instead of a string for these fields? Maybe there are other ways we could help.

Hi @ojroques. I create the kubernetes cluster via Terraform. I also create the credentials to the Scaleway Secret Manager via Terraform, which then I store them in a Kubernetes Secret.

I configure the SecretStore using this secret, but I deploy this - and other kubernetes resources - using a GitOps operator (Fleet, Rancher). And the GitOps operator does not have any knowledge of the Scaleway Secret Manager, nor project id. The access key and secret key are configurable (which I get from the secretRef).

However, the region and projectId are not referenceable. This way I need to hard code the project ID and region in each of the environments. Which requires duplication of this yaml resource per environment, where only the project ID changes. This is more human error prone and necessary duplication. Not a big problem but might cause some issues.

**Comment by ojroques** [comment]
I see. We cannot update the type of the `region` and `projectID` arguments since it would break things.

We could add new arguments like `projectIDfromSecret` but it would complexify the configuration for not much benefits. But if more users face the same issue, we'll definitely add it.

**Comment by mig4ng** [comment]
@ojroques I agree we should not break things nor increase the complexity of the configuration. But this would indeed be a nice to have.

An alternative could be having 2 providers `scaleway` and `scaleway-dynamic` (or another name) which allowed this. We could then see which provider users gravitate to. But this too has it's own cons.

**Comment by evrardj-roche** [comment]
This issue has been marked as Stale and will be closed soon if there is no activity. Please update the issue or engage with it to keep it open. It is fine to re-open this issue in the future if you want to work on this.

**Comment by evrardj-roche** [comment]
Hello. Thank you for taking the time to open this issue and contribute to the project.
We truly appreciate community involvement and the effort that goes into reporting issues and suggesting improvements.

We're closing this (Stale) issue as part of a maintenance effort to keep our issue tracker focused and actionable.
Issues that have been inactive for an extended period can create clutter, make it harder to prioritize current work, and distract maintainers from issues that have active contributors ready to work on them.

This doesn't mean your issue isn't valid or important.
If you or anyone else is still affected by this issue and willing to contribute toward a solution (whether through discussion, testing, or code),
please feel free to comment here, reopen the issue, and mention you want to be assigned on this issue.

Alternatively, you can directly submit a PR with `Closes <PR NUMBER>` in its content.

We're happy to reopen or revisit this if there's renewed interest and someone willing to champion it forward.

Thanks again for being part of our community!

**Comment by mig4ng** [comment]
Still relevant
